sql/src/main/java/io/crate/planner/operators/WindowAgg.java

```diff
@@ -146,27 +144,52 @@ public ExecutionPlan build(PlannerContext plannerContext,
             functionsWithInputs.put(windowFunctionSymbol, inputs);
         }
 
-        OrderBy orderBy = windowDefinition.orderBy();
-        int[] orderByIndexes = EMPTY_ORDER_BY_INDEXES;
-        if (orderBy != null) {
-            InputColumns.SourceSymbols orderByCtx = new InputColumns.SourceSymbols(source.outputs());
-            List<Symbol> outputs = InputColumns.create(source.outputs(), orderByCtx);
-            List<Symbol> orderByInputColumns = InputColumns.create(orderBy.orderBySymbols(), orderByCtx);
-
-            orderByIndexes = new int[orderByInputColumns.size()];
-            for (int i = 0; i < orderByInputColumns.size(); i++) {
-                Symbol orderBySymbol = orderByInputColumns.get(i);
-                assert orderBySymbol instanceof InputColumn : "Window ordering should be expressed as ICs at this stage";
-                orderByIndexes[i] = ((InputColumn) orderBySymbol).index();
+        OrderBy orderBy = windowDefinitionWithICs.orderBy();
+        List<Symbol> orderByICs = new ArrayList<>();
+        if (!windowDefinition.partitions().isEmpty() || orderBy != null) {
```

Comment:
I wonder if we could extract some logic here so that it looks a bit more declarative. (At least within the method here). Maybe by having some  `OrderBy.concat` functionality?

Reply:
Yeah, not sure, it bothers me as well a bit, as the `OrderBy` is optional (nullable). I don't see a clear (and clean) logical abstraction here

Reply:
Hmm, thinking about it some more, do you think it would be better to create an `OrderBy` object that'll be used for the `OrderedTopNProjection` ? (we could at least extract a method inside the `WindowAgg` that creates this `OrderBy` with the symbols being the partitionby symbols + orderby sybols)

Reply:
Would be an option I guess. It's also okay for me to leave it as is and follow up on it later.